test/Libraries/Microsoft.Extensions.AI.Integration.Tests/ChatClientIntegrationTests.cs

```diff
@@ -163,13 +163,25 @@ public virtual async Task FunctionInvocation_AutomaticallyInvokeFunction_Paramet
 
         int secretNumber = 42;
 
-        var response = await chatClient.CompleteAsync("What is the current secret number?", new()
+        List<ChatMessage> messages =
+        [
+            new(ChatRole.User, "What is the current secret number?")
+        ];
+
+        var response = await chatClient.CompleteAsync(messages, new()
         {
             Tools = [AIFunctionFactory.Create(() => secretNumber, "GetSecretNumber")]
         });
 
         Assert.Single(response.Choices);
         Assert.Contains(secretNumber.ToString(), response.Message.Text);
+
+        if (response.Usage is { } finalUsage)
+        {
+            UsageContent? intermediate = messages.SelectMany(m => m.Contents).OfType<UsageContent>().FirstOrDefault();
+            Assert.NotNull(intermediate);
+            Assert.True(finalUsage.TotalTokenCount > intermediate.Details.TotalTokenCount);
```

Comment:
I must be missing something - can you clarify why `finalUsage.TotalTokenCount > intermediate.Details.TotalTokenCount`?

I would have thought the final message token count was independent of the token count on preceding messages.

Reply:
Oh is the point that `TotalTokenCount` includes input tokens as well as output tokens, and the input for the final message includes the output from preceding messages?

Reply:
> Oh is the point that `TotalTokenCount` includes input tokens as well as output tokens, and the input for the final message includes the output from preceding messages?

Yup. So the first request sends A tokens and the response has B tokens, then the next request include A and B tokens (since they're in the history) plus an additional C tokens, and has a response with D tokens. As long as C and D aren't 0, the final total should be greater than the intermediate total.

I think... right?

Reply:
Sounds correct!